test/role_test.js

```diff
@@ -138,6 +138,29 @@ helper.secrets.mockSuite(helper.suiteName(__filename), ['app', 'azure'], functio
     assert(roles.some(role => role.roleId === 'thing-id:' + clientId));
   });
 
+  test('listRoleIds', async () => {
+    let result = await helper.apiClient.listRoleIds();
+    assert(result.roleIds.some(roleId => roleId === 'thing-id:' + clientId));
```

Comment:
It'd be good to assert that you get the whole list here.

Reply:
How should I do that? I'll need the actual length of the list right?

Reply:
Hm, good question.  How many roles are there at this point?  It looks like this test suite does not clear the list of roles between test executions, so what's left at this point is whatever the previous tests have created.

You could reset the roles to an empty list with
```js
await this.Roles.modify(roles => ([]));
```
and then fill it with a set of roles by calling createRole a few times in a loop.  That would, at least, give you a stable set of roles to test against.

Reply:
this.Roles seems to be undefined. I'm getting this

`TypeError: Cannot read property 'modify' of undefined`

Reply:
Oops, sorry, `helper.Roles`

Reply:
Ah, good fix.